File: 01/loop.py
#!/usr/bin/env python3
# -*- coding : UTF-8 -*-

# loop.py
# 実行時間の比較（Python版）

import random  # 乱数のライブラリを使うため

MAXCOUNT = 100000000  # ループ回数(1億回)
count = 0   # 結果を入れる変数。0で初期化

print("Python: start")

for i in range(MAXCOUNT):
    if random.randint(0, 255) % 2 == 0: # 偶数の場合
        count -= 1                      # 1減らす
    else:                               # 奇数の場合
        count += 1                      # 1増やす

print(f"num = {count}")

File: 01/loop.c
// loop.c
// 実行時間の比較（C言語版）

#include <stdio.h>      // printf()などを使うのに必要
#include <stdlib.h>     // random()などを使うのに必要
#include <time.h>       // time()などを使うのに必要

#define MAXCOUNT 100000000   // ループ回数(1億回)

int main(void) {
    int count = 0;  // 結果を入れる。0で初期化
    long i;         // ループ変数

    srandom(time(NULL));  // 乱数のシードを初期化

    printf("C: start\n");

    for (i = 0; i < MAXCOUNT; i++) {
        if (random() % 2 == 0) { // 偶数の場合
            count -= 1;          // 1減らす
        } else {                 // 奇数の場合
            count += 1;          // 1増やす
        }
    }

    printf("num = %d\n", count); // 結果を表示

    return 0;
}
